lib/Sema/SemaBounds.cpp

```diff
@@ -374,12 +392,26 @@ namespace {
     }
 
     // Compute bounds for the target of an lvalue.  Values assigned through
-    // the lvalue must meet satisfy these bounds.   Values read through the
+    // the lvalue must satisfy these bounds.   Values read through the
     // lvalue will meet these bounds.
     BoundsExpr *LValueTargetBounds(Expr *E) {
       assert(E->isLValue());
       // TODO: handle side effects within E
       E = E->IgnoreParens();
+      QualType QT = E->getType();
+
+      // If the target value v is a Ptr type, it has bounds(v, v + 1), unless
+      // it is a function pointer type, in which case it has no required
+      // bounds.
+      if (QT->isCheckedPointerPtrType()) {
+         if (QT->isFunctionPointerType())
+           return CreateBoundsNone();
```

Comment:
Shouldn't function pointers have a count(1) bounds or equivalent, matching any general `ptr<>`. 

My understanding of the semantics is that it is an error to dereference `bounds(none)`, which is arguably what happens when we call a function. 

This also goes for `bounds(none)` below where we apply the deref to a function pointer.